src/mame/video/tmap038.h

```diff
@@ -0,0 +1,105 @@
+
+	// configurations
+	template <typename T> void set_gfxdecode_tag(T &&tag) { m_gfxdecode.set_tag(std::forward<T>(tag)); }
+	typedef device_delegate<void (bool tiledim, u32 &color, u32 &pri, u32 &code)> tmap038_cb_delegate;
+	void set_tile_callback(tmap038_cb_delegate cb) { m_038_cb = cb; }
+	void set_gfx(u16 no) { m_gfxno = no; }
+
+	// call to do the rendering etc.
+	template<class _BitmapClass>
+	void draw_common(screen_device &screen, _BitmapClass &bitmap, const rectangle &cliprect, u32 flags, u8 pri = 0, u8 pri_mask = ~0);
+
+	void prepare();
+	void draw(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect, u32 flags, u8 pri = 0, u8 pri_mask = ~0);
+	void draw(screen_device &screen, bitmap_rgb32 &bitmap, const rectangle &cliprect, u32 flags, u8 pri = 0, u8 pri_mask = ~0);
+
+	// access
+	void vram_map(address_map &map);
+	void vram_8x8_map(address_map &map);
+	void vram_16x16_map(address_map &map);
+
+	void vram_writeonly_map(address_map &map);
+	void vram_16x16_writeonly_map(address_map &map);
+
+	u16 vram_8x8_r(offs_t offset);
+	void vram_8x8_w(offs_t offset, u16 data, u16 mem_mask);
+
+	u16 vram_16x16_r(offs_t offset);
+	void vram_16x16_w(offs_t offset, u16 data, u16 mem_mask);
+
+	u16 lineram_r(offs_t offset) { return m_lineram[offset]; }
+	void lineram_w(offs_t offset, u16 data, u16 mem_mask = ~0) { COMBINE_DATA(&m_lineram[offset]); }
+ 
+	u16 vregs_r(offs_t offset) { return m_vregs[offset]; }
+	void vregs_w(offs_t offset, u16 data, u16 mem_mask = ~0) { COMBINE_DATA(&m_vregs[offset]); }
+ 
+	void mark_all_dirty() { m_tmap->mark_all_dirty(); };
+	void set_flip(u32 attributes) { m_tmap->set_flip(attributes); }
+	void set_palette_offset(u32 offset) { m_tmap->set_palette_offset(offset); }
+	void set_scroll_rows(u32 scroll_rows) { m_tmap->set_scroll_rows(scroll_rows); }
+	void set_scroll_cols(u32 scroll_cols) { m_tmap->set_scroll_cols(scroll_cols); }
+	void set_scrollx(int which, int value) { m_tmap->set_scrollx(which, value); }
+	void set_scrolly(int which, int value) { m_tmap->set_scrolly(which, value); }
+
+	// getters
+	u16 lineram(offs_t offset) { return m_lineram[offset]; }
+
+	u16 rowscroll(offs_t line) { return rowscroll_en() ? m_lineram[((line & 0x1ff) * 2) + 0] : 0; }
+	u16 rowselect(offs_t line) { return rowselect_en() ? m_lineram[((line & 0x1ff) * 2) + 1] : 0; }
+
+	u16 vregs(offs_t offset) { return m_vregs[offset]; }
+
+	bool flipx()        { return BIT(~m_vregs[0], 15); }
+	bool rowscroll_en() { return BIT(m_vregs[0], 14) && (m_lineram != nullptr); }
+	u16 scrollx()       { return m_vregs[0] & 0x1ff; }
+
+	bool flipy()        { return BIT(~m_vregs[1], 15); }
+	bool rowselect_en() { return BIT(m_vregs[1], 14) && (m_lineram != nullptr); }
+	bool tiledim()      { return m_tiledim; }
+	u16 scrolly()       { return m_vregs[1] & 0x1ff; }
+
+	bool enable()       { return BIT(~m_vregs[2], 4); }
+	u16 external()      { return m_vregs[2] & 0xf; }
```

Comment:
You should `const`-qualify accessors that aren't supposed to change state, e.g. `bool flipx() const { return ...; }`